File: javapatterns/src/main/java/org/iut/nantes/state/car/Conduite.java
package org.iut.nantes.state.car;

class Conduite extends Etat
{
  public Conduite( GoogleCar car )
  {
    super( car );
  }

  public void arrive()
  {
    System.out.println("Arrivé à destination : on repasse en état d'attente.");
    car.setEtat( car.getAttente() );
  }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/state/car/etats/Conduite.kt
package org.iut.nantes.state.car.etats

import org.iut.nantes.state.car.Car

class Conduite(car: Car) : Etat(car) {

    override fun action() {
        println("Vroum vroum ! ")
        car.changeEtatAttente()
    }

}